### PR TITLE
Modify IdP developer docs

### PR DESCRIPTION
**Pull request by royalchan-okta**
1. Moved credentials object into the protocol object.
2. Updated IdP Key docs

### CODE REVIEW

**Comment by royalchan-okta** [comment]
@federations-okta @karlmcguinness-okta @hassenkaraa-okta

File: _source/docs/api/resources/idps.md [hunk 1]
```diff
@@ -163,13 +162,14 @@ IdP-specific protocol settings for endpoints, bindings, and algorithms used to c
...
+| ----------- | ------------------------------------------------------------------ | -------------------------------------------------------- | -------- | -------- | --------- | --------- | ---------- |
+| type        | SAML 2.0 protocol                                                  | `SAML2`                                                  | FALSE    | TRUE     |           |           |            |
+| endpoints   | SAML 2.0 HTTP binding settings for IdP and SP (Okta)               | [SAML 2.0 Endpoints Object](#saml-20-endpoints-object)   | FALSE    | FALSE    |           |           |            |
+| algorithms  | Settings for signing and verifying SAML messages                   | [SAML 2.0 Algorithms Object](#saml-20-algorithms-object) | FALSE    | FALSE    |           |           |            |
+| credentials | Federation trust credentials for verifying assertions from the IdP | [Credentials Object](#saml-20-trust-credentials-object)  | FALSE    | FALSE    |           |           |            |
```
Comment (karlmcguinness-okta) [inline _source/docs/api/resources/idps.md]: [SAML 2.0 Credentials Object](#saml-20-trust-credentials-object)

File: _source/docs/api/resources/idps.md [hunk 2]
```diff
@@ -413,17 +420,45 @@ XML digital signature algorithm settings for verifying `<SAMLResponse>` messages
...
+| Property | Description                                                                                   | DataType | Nullable | Readonly | MinLength | MaxLength | Validation                                 |
+| -------- | --------------------------------------------------------------------------------------------- | -------- | -------- | -------- | --------- | --------- | ------------------------------------------ |
+| issuer   | URI that identifies the issuer (IdP) of a SAML `<SAMLResponse>` message `<Assertion>` element | String   | FALSE    | FALSE    | 1         | 1024      | [URI](https://tools.ietf.org/html/rfc3986) |
+| audience | URI that identifies the target Okta IdP instance (SP) for an `<Assertion>`                    | String   | FALSE    | FALSE    | 1         | 1024      | [URI](https://tools.ietf.org/html/rfc3986) |
+| kid      | Key ID reference to the IdP's X.509 signature certificate                                     | String   | FALSE    | FALSE    | 36        | 36        | Valid IdP Key ID reference                 |
```
Comment (karlmcguinness-okta) [inline _source/docs/api/resources/idps.md]: Can we hyperlink Key ID reference to Key Object?

File: _source/docs/api/resources/idps.md [hunk 3]
```diff
@@ -529,14 +607,44 @@ The `OIDC` protocol supports the `authorization` and `token` endpoints
...
+      "token": {
+        "url": "https://www.googleapis.com/oauth2/v3/token",
+        "binding": "HTTP-POST"
+      }
+    }
+  }
+}
+~~~
+
+##### Openid Connect Client Credentials Object
```
Comment (karlmcguinness-okta) [inline _source/docs/api/resources/idps.md]: Thought... may we should consolidate the OAuth and OIDC Endpoints and Credentials Objects?  Not sure we need 2 types for each, just need to clear up the eternal confusion of OAuth vs OIDC :)

Comment (royalchan-okta) [inline _source/docs/api/resources/idps.md]: Sure, they really don't differ. Should we just title and caption everything as OAuth 2.0/Openid Connect?

File: _source/docs/api/resources/idps.md (hunk 2 above)
Comment (royalchan-okta) [inline _source/docs/api/resources/idps.md]: There isn't a Key Object entry, linking to the operations

**Comment by royalchan-okta** [comment]
@mattloughran-okta @joelfranusic-okta

**Comment by mattloughran-okta** [comment]
Looks good!

**Pull request merged**